### PR TITLE
HMAN-156 updated partyRole to have 40 characters

### PR DESCRIPTION
**Pull request by CameronGuthrie1**
### JIRA link (if applicable) ###

[HMAN-156 - Change partyRole maxLength in the PUT/POST hearing request payload](https://tools.hmcts.net/jira/browse/HMAN-156)

### Change description ###
Updated partyRole in the partyDetails to allow 40 characters

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by danlysiak — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java [hunk 1]
```diff
@@ -507,4 +508,37 @@ void whenInvalidPartyRoleIsEmpty() {
...
 
+    @Test
+    void whenValidPartyRoleIs40Characters() {
+        PartyDetails partyDetails = new PartyDetails();
+        partyDetails.setPartyID("XXX1");
+        partyDetails.setPartyType(PartyType.IND.getLabel());
+        partyDetails.setPartyRole("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMN");
+        Set<ConstraintViolation<PartyDetails>> violations = validator.validate(partyDetails);
+        assertTrue(violations.isEmpty());
+        List<String> validationErrors = new ArrayList<>();
```
Comment (sullivangareth) [inline src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java:519]: Don't need lines 519 - 524? Finish tests once we assert violations is empty

File: src/main/java/uk/gov/hmcts/reform/hmc/model/PartyDetails.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public class PartyDetails {
     private String partyType;
 
     @NotEmpty(message = ValidationError.PARTY_ROLE_EMPTY)
-    @Size(max = 6, message = ValidationError.PARTY_ROLE_MAX_LENGTH)
+    @Size(max = 40, message = ValidationError.PARTY_ROLE_MAX_LENGTH)
```
Comment (sullivangareth) [inline src/main/java/uk/gov/hmcts/reform/hmc/model/PartyDetails.java:29]: You will need to change the value of `ValidationError.PARTY_ROLE_MAX_LENGTH` from

`public static final String PARTY_ROLE_MAX_LENGTH = "Party role must not be more than 6 " + CHARACTERS_LONG;`

to

`public static final String PARTY_ROLE_MAX_LENGTH = "Party role must not be more than 40 " + CHARACTERS_LONG;`

File: src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java [hunk 3]
```diff
@@ -507,4 +508,37 @@ void whenInvalidPartyRoleIsEmpty() {
...
+    }
+
+    @Test
+    void whenInvalidPartyRoleIs41Characters() {
+        PartyDetails partyDetails = new PartyDetails();
+        partyDetails.setPartyID("XXX1");
+        partyDetails.setPartyType(PartyType.IND.getLabel());
+        partyDetails.setPartyRole("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNO");
+        Set<ConstraintViolation<PartyDetails>> violations = validator.validate(partyDetails);
+        assertFalse(violations.isEmpty());
```
Comment (sullivangareth) [inline src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java:534]: Could reduce lines 534 - 541 down to

```
assertEquals(1, violations.size());

assertTrue(violations.stream()
    .map(ConstraintViolation::getMessage)
    .anyMatch(msg -> msg.equals(PARTY_ROLE_MAX_LENGTH)));
```

File: src/main/java/uk/gov/hmcts/reform/hmc/model/PartyDetails.java (hunk 2 above)
Comment (CameronGuthrie1) [reply src/main/java/uk/gov/hmcts/reform/hmc/model/PartyDetails.java:29]: done

File: src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java (hunk 1 above)
Comment (CameronGuthrie1) [reply src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java:519]: done

File: src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java (hunk 3 above)
Comment (CameronGuthrie1) [reply src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java:534]: done

File: src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java (hunk 1 above)
Comment (CameronGuthrie1) [reply src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java]: Updated in 96433765a84f6a5fe58bb8429db9c033c06eeaeb

File: src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java (hunk 3 above)
Comment (CameronGuthrie1) [reply src/test/java/uk/gov/hmcts/reform/hmc/validator/BeanValidatorTest.java]: Updated in 96433765a84f6a5fe58bb8429db9c033c06eeaeb

**Review by sullivangareth — APPROVED** [review]
